frontend/model/contracts/group.js

```diff
@@ -1154,15 +1154,18 @@ sbp('chelonia/defineContract', {
                       })
                     }
                   }
-                }).catch((e) => {
+                }).catch(async (e) => {
                   // If already joined, ignore this error
                   if (e?.name === 'GIErrorUIRuntimeError' && e.cause?.name === 'GIGroupAlreadyJoinedError') return
-                  console.error('Error while joining the #General chatroom', e);
-                  // avoid blocking the main thread
-                  // eslint-disable-next-line require-await
-                  (async () => {
-                    alert(L("Couldn't join the #{chatroomName} in the group. An error occurred: #{error}.", { chatroomName: CHATROOM_GENERAL_NAME, error: e?.message || e }))
-                  })()
+                  console.error('Error while joining the #General chatroom', e)
+                  const errMsg = L("Couldn't join the #{chatroomName} in the group. An error occurred: {error}", { chatroomName: CHATROOM_GENERAL_NAME, error: e?.message || e })
+                  const promptOptions = {
+                    heading: L('Error while joining a chatroom'),
+                    question: errMsg,
+                    primaryButton: L('Close')
+                  }
+
+                  await sbp('gi.ui/prompt', promptOptions)
```

Comment:
If you look at how it was previously, it was written in a non-blocking way, which I believe is no longer the case. One reason not to await here is that now you've made an error handler that could potentially throw. Ideally, error handlers should _not_ result in an error.

I'd make it non-blocking again because ideally contracts should not have code that depends on user interation, even when it doesn't block message processing. In this case, I think you can resolve it by removing the `await` and adding a `.catch()` handler (this is because not using a catch handler would result in an unhandled promise rejection, if the promise throws).

Reply:
@corrideat fair point. I think we can just remove `async / await` here, because there is no required action followed by clicking on that 'Close' button on the prompt.